audtorch/datasets/musdb18.py

```diff
@@ -0,0 +1,172 @@
+import os
+import random
+
+import numpy as np
+import musdb
+from torch.utils.data import Dataset
+
+from .base import _include_repr
+from .utils import sampling_rate_after_transform
+
+
+__doctest_skip__ = ['*']
+
+
+class Musdb18(Dataset):
+    r"""Musical Source Separation musdb18 data set.
+
+    Data set used in the SiSEC MUS challange 2018:
+    https://sigsep.github.io/datasets/musdb.html
+
+    License: restricted to research, varies for different tracks.
+
+    `musdb18` contains 150 full length music tracks (~10 hours) of different
+    styles along with their isolated `drums`, `bass`, `vocals`, and `others`
+    setms.
+
+    It is split into a `train` subset containing 100 songs, and a `test` subset
+    containing the remaining 50 songs. Supervised approaches should be trained
+    on the training set and tested on both sets.
+
+    `musdb18` is composed by 100 tracks from the `DSD100 dataset`_, 46 tracks
+    from MedleyDB_, 2 tracks from Native Instruments, and 2 tracks from the
+    `heise stems remix competition`_.
+
```

Comment:
I think this needs a bit more comments on what else this class supports. For example:

* It supports returning just the accompaniment with a silent target
* It supports applying a common transform on both signals and two different ones on each of them afterwards

In addition, it would be nice if you could explain that this is a wrapper over an existing data set that extends in a way that makes it suitable for training.

Reply:
I added a little bit more documentation on `joint_transform` and `percentage_silence`.

What do you mean with
> this is a wrapper over an existing data
All our classes are wrappers?